pkg/neobench/worker.go

```diff
@@ -108,21 +108,56 @@ func (w *Worker) gatherResults(workloadStats map[string]*ScriptResult, workStart
 
 func (w *Worker) runUnit(session neo4j.Session, uow UnitOfWork) uowOutcome {
 	transaction := func(tx neo4j.Transaction) (interface{}, error) {
+		var lastResult neo4j.Result
+
 		for _, s := range uow.Statements {
 			res, err := tx.Run(s.Query, s.Params)
 			if err != nil {
 				return nil, err
 			}
-			_, err = res.Consume()
+			_, err = res.(neo4j.Result).Consume()
+			if err != nil {
+				return nil, err
+			}
+			lastResult = res
+		}
+		return lastResult, nil
+	}
+
+	autocommitTransaction := func(session neo4j.Session) (interface{}, error) {
+		var lastResult neo4j.Result
+		var retries = 100
+		var res interface{}
+		var err error
+
+		for _, s := range uow.Statements {
+			var retriesThisTime = retries
+			for i := 0; i < retriesThisTime; i++ {
+				res, err = session.Run(s.Query, s.Params)
```

Comment:
Is there a RunRead or otherwise a means to make the autocommit transactions route to followers if `uow.Readonly`? Otherwise there's some sneakyness here right, if someone later on compares tx-functions vs autocommit, and don't realize that they both change to autocommit *and* change to route everything to one instance.

Reply:
There is no run-read. There totally should be. You can use bolt:// with Run to do a run-read but I think that's out of scope here.

Thinking about it we should always read using readTransaction and only do the run (auto commit) transaction for writes